stdlib/public/SwiftShims/RuntimeShims.h

```diff
@@ -84,6 +84,32 @@ __swift_size_t _swift_stdlib_getHardwareConcurrency(void);
 /// the Swift runtime and user binaries need to agree on this value.
 #define _swift_MinAllocationAlignment (__swift_size_t) 16
 
+struct MetadataSectionRange {
+  __swift_uintptr_t start;
+  __swift_size_t length;
+};
+
+struct MetadataSections {
```

Comment:
@mikeash I talked with @compnerd yesterday and he suggested we reuse the `MetadataSections` struct (instead of defining a brand new one), which was defined in both `ImageInspectionElf.h` and `ImageInspectionCOFF.h`, so it's accessible from Swift as well. He suggested I place it under /include/swift/Runtime, and confirm with you it that'd be the right place. So would that be the best place? And if yes, how can I expose data structures defined in /include/swift/Runtime so they're accessible from Swift? I tried yesterday but I couldn't figure it out.

Reply:
@mikeash Thank you! Follow up question: is it possible to make a C++ struct visible to Swift? Or do I need to declare it inside an extern block (as a C struct)?

Reply:
Anything visible to Swift will be parsed as plain C, so the contents of your new file need to be C compatible. The other files in there do namespacing and other C++isms inside `#ifdef __cplusplus` blocks so that they can be parsed either way.